Likhovodov/Simulated-Conversations
# Adding audio support for conversations

To test:

- Create everything necessary for a conversation template and assign it to student
- From the student home page, start the assignment
- On each node you will see a video and some audio control buttons
- Once record is pressed, you will be asked (only once) for permission to your mic. It then immediately starts recording...
Might add a "Recording..." text to signal that it's currently recording.
- Hit stop, and video will disappear and choices will pop-up.
- Rinse and repeat

Notes:
- UI is a little janky in that you can, under specific circumstances, go back and record a new response and choose a different choice; however, this data doesn't persist. Only the original audio recording and original choice that you make will persist.
- If you are able to record a second audio clip, you will need to hit refresh on the choice page in order to hear the original audio.
- I've created an 'exit conversation' button for the user to click if they wish to quit mid-conversation. Would like to trigger a modal warning that their response will be marked as incomplete. Will create a ticket for this feature.
